tfjs-core/src/kernel_registry_test.ts

```diff
@@ -129,22 +129,25 @@ describeWithFlags('kernel_registry', ALL_ENVS, () => {
     expect(lastStorageId).toBe(-1);
 
     // Kernel was executed on the first backend.
-    tf.setBackend('backend1');
+    await tf.setBackend('backend1');
     tf.engine().runKernel('MyKernel', {}, {});
     expect(lastStorageId).toBe(1);
 
     // Kernel was executed on the second backend.
-    tf.setBackend('backend2');
+    await tf.setBackend('backend2');
     tf.engine().runKernel('MyKernel', {}, {});
     expect(lastStorageId).toBe(2);
 
     tf.removeBackend('backend1');
     tf.removeBackend('backend2');
     tf.unregisterKernel('MyKernel', 'backend1');
     tf.unregisterKernel('MyKernel', 'backend2');
+
+    // Revert backend mutation.
+    await tf.setBackend(testEnv.backendName);
```

Comment:
I think it's better to reset the backend in `afterEach`. Right now, if an error is thrown in this test, the backend will not be reset. 

Alternatively, you can wrap the entire test in `try { [test] } finally { [reset backend] }`.